Makefile

```diff
@@ -0,0 +1,144 @@
+SHELL = /bin/bash
+
+.DEFAULT_GOAL := help
+
+
+CURRENT_DIR := $(shell pwd)
+VENV := $(CURRENT_DIR)/.venv
+DEV_REQUIREMENTS = $(CURRENT_DIR)/dev_requirements.txt
+TEST_REPORT_DIR := $(CURRENT_DIR)/tests/report
+TEST_REPORT_FILE := nose2-junit.xml
+
+# venv targets timestamps
+VENV_TIMESTAMP = $(VENV)/.timestamp
+DEV_REQUIREMENTS_TIMESTAMP = $(VENV)/.dev-requirements.timestamp
+
+# general targets timestamps
+TIMESTAMPS = .timestamps
+PREP_PACKAGING_TIMESTAMP = $(TIMESTAMPS)/.prep-packaging.timestamp
+
+# Find all python files that are not inside a hidden directory (directory starting with .)
+PYTHON_FILES := $(shell find ./* -type d \( -path ./build -o -path ./dist \) -prune -false -o -type f -name "*.py" -print)
+
+PYTHON_VERSION := 3
+SYSTEM_PYTHON := python3
+
+# Commands
+PYTHON := $(VENV)/bin/python3
+PIP := $(VENV)/bin/pip3
+YAPF := $(VENV)/bin/yapf
+NOSE := $(VENV)/bin/nose2
+PYLINT := $(VENV)/bin/pylint
+
+PACKAGE_VERSION = $(shell awk '/^Version:/ {print $$2}' logging_utilities.egg-info/PKG-INFO)
+
+
+all: help
+
+
+.PHONY: help
+help:
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Possible targets:"
+	@echo -e " \033[1mSetup TARGETS\033[0m "
+	@echo "- setup              Create the python virtual environment with developper tools"
+	@echo -e " \033[1mFORMATING, LINTING AND TESTING TOOLS TARGETS\033[0m "
+	@echo "- format             Format the python source code"
+	@echo "- lint               Lint the python source code"
+	@echo "- format-lint        Format and lint the python source code"
+	@echo "- test               Run the tests"
+	@echo -e " \033[1mPACKAGING TARGETS\033[0m "
+	@echo "- package            Create package"
+	@echo "- publish            Tag and publish package to PyPI"
+	@echo -e " \033[1mCLEANING TARGETS\033[0m "
+	@echo "- clean              Clean genereated files"
+	@echo "- clean_venv         Clean python venv"
+
+
+# Build targets. Calling setup is all that is needed for the local files to be installed as needed.
+
+.PHONY: setup
+setup: $(DEV_REQUIREMENTS_TIMESTAMP)
+
+
+# linting target, calls upon yapf to make sure your code is easier to read and respects some conventions.
+
+.PHONY: format
+format: $(DEV_REQUIREMENTS_TIMESTAMP)
+	$(YAPF) -p -i --style .style.yapf $(PYTHON_FILES)
+
+
+.PHONY: lint
+lint: $(DEV_REQUIREMENTS_TIMESTAMP)
+	$(PYLINT) $(PYTHON_FILES)
+
+
+.PHONY: format-lint
+format-lint: format lint
+
+
+# Test target
+
+.PHONY: test
+test: $(DEV_REQUIREMENTS_TIMESTAMP)
+	mkdir -p $(TEST_REPORT_DIR)
+	$(NOSE) -c tests/unittest.cfg --plugin nose2.plugins.junitxml --junit-xml --junit-xml-path $(TEST_REPORT_DIR)/$(TEST_REPORT_FILE) -s tests/
+
+
+# Packaging target
+
+.PHONY: package
+package: $(PREP_PACKAGING_TIMESTAMP)
+	python3 setup.py sdist bdist_wheel
+
+
+.PHONY: publish
+publish: clean test package
+	@echo "Tag and upload package version=$(PACKAGE_VERSION)"
+	@if [ -n "`git status --porcelain`" ]; then echo "Repo is dirty !"; exit 1; fi
+	python3 -m twine upload dist/*
```

Comment:
Well because the repository is public and have actually nothing to do with swisstopo (general logging library) where people that have nothing to do with us could contribute, I though it would not be a good idea to put internal documentation stuff inside this repo.
I was more thinking of a general PyPI documentation on `doc-guideline` repository on how to publish packages.
About the automated publishing mechanism I don't expect much versions (2-3 versions after initial one for the next coming years) and thought that doing automation here would be overkill.

Reply:
As I said, I'm very much in favor of a non automated publishing process.

Documentation in doc guideline is  a good idea and works perfectly fine for me.